src/Traversals/bfs.jl

```diff
@@ -80,15 +81,50 @@ end
 end
 
 """
-    distances(g, s, alg=BFS())
-    distances(g, ss, alg=BFS())
+    distances(g, s, alg=BreadthFirst())
+    distances(g, ss, alg=BreadthFirst())
 
 Return a vector filled with the geodesic distances of vertices in  `g` from vertex `s` / unique vertices `ss`
-using BFS traversal algorithm `alg`.  For vertices in disconnected components the default distance is `typemax(T)`.
+using BreadthFirst traversal algorithm `alg`.  For vertices in disconnected components the default distance is `typemax(T)`.
 """
-function distances(g::AbstractGraph{T}, s, alg::BFS=BFS()) where T
+function distances(g::AbstractGraph{T}, s, alg::BreadthFirst=BreadthFirst()) where T
     d = fill(typemax(T), nv(g))
     state = DistanceState(d, one(T))
     traverse_graph!(g, s, alg, state)
     return state.distances
 end
+
+mutable struct PathState{T} <: AbstractTraversalState
```

Comment:
Is `T` Integer? In that case maybe subtyping could help

Reply:
T is integer because `vertices` is Integer. Is there a reason to limit it other than documentation?

Reply:
you don't have it in the members of PathState, which looks odd, you could create a `PathState{String}` given the current fields, which is bizarre if you explicitly know it will only be integers

Reply:
done.